extensions/websockets-next/runtime/src/main/java/io/quarkus/websockets/next/runtime/BasicWebSocketConnectorImpl.java

```diff
@@ -144,7 +143,7 @@ public Uni<WebSocketClientConnection> connect() {
             throw new WebSocketClientException(e);
         }
 
-        return UniHelper.toUni(client.connect(connectOptions))
+        return Uni.createFrom().completionStage(() -> client.connect(connectOptions).toCompletionStage())
```

Comment:
Should we replace the lambda with an anonymous class?

Reply:
Always :)(

Reply:
I'm sorry but no. The reactive code is difficult to read even with lambdas. Also we use lambdas everywhere in this extension. How do you determine where to use lambda and where an anonymous class?  I know that we used to avoid lambdas where possible due to some memory overhead but (1) I'd like to know if it's still the case for new JDKs and (2) I don't want to spend too much time on performance until we have some reasonable benchmarks.

Reply:
> -1 for everyrhing mentioned above

Any arguments?

Reply:
The argument is simple: It's been a (loose) policy all along, so if we want to change it, we should do the bare minimum of trying to figure out whether it makes sense to or not, not just put forth generic arguments that could be applied to almost any situation.

Reply:
Fair enough